### DISCUSSION TITLE
Unresolved References

### DISCUSSION THREAD

**Discussion by mohamadreza77** [eclipse-epsilon/epsilon#109] [Q&A] [answered]
Hello there!

I developed a simple Sirius VSM and named it 'extendedfamily.odesign'. It includes a relation-based edge with two different nodes as its sourceMapping and targetMapping. I noticed that it uses special addressing for these references, for example:

`targetMapping="//@ownedViewpoints[name='FirstViewpoint']/@ownedRepresentations[name='Persons%20Diagram']/@defaultLayer/@nodeMappings[name='ManNode']"`

I aimed to load this model in memory using Java APIs in a way that conforms to the Sirius Description Metamodel. Everything seemed fine until I encountered the following error:

> Unresolved reference '//@ownedViewpoints[name='FirstViewpoint']/@ownedRepresentations[name='Persons%20Diagram']/@defaultLayer/@nodeMappings[name='WomanNode']'.

I would appreciate any help on how to make the model understand this referencing method. Any suggestions?

Thanks!

**Comment by kolovos** [comment]
Hi Mohamadreza,

Could you please elaborate a bit more on how this question is related to Epsilon and provide a [minimal example](https://eclipse.dev/epsilon/doc/articles/minimal-examples/) we can use to reproduce this issue?

Thanks,
Dimitris

**Reply by mohamadreza77** [reply]
Thank you for your reply. So, I want to transform the VSM into another model using ETL. I am working with Java APIs. I have to provide a metamodel (the Sirius Metamodel) so that Epsilon would know that my odesign file conforms to that metamodel.

To do that, we have to register our metamodel with Epsilon. I created a Sirius metamodel (a simple version similar to thee Sirius Metamodel) using Ecore. Then, I registered that ecore metamodel and defined the namespaces in Java. When Epsilon wants to load the odesign model, it seems that it cannot resolve the references defined in edgeMapping. What am I missing? Isn't there a need to define a Sirius metamodel?

**Reply by kolovos** [reply]
Could you please share a [minimal version](https://eclipse.dev/epsilon/doc/articles/minimal-examples/) of the code that's giving you this error in a zip attachement (or preferrably a small GitHub repository) so that we can have a look and advise?

**Comment by kolovos** [comment]
In case it helps, to successfully load Sirius odesign models from Java in a side project, we had to set the resource's `SIRIUS_USE_URI_FRAGMENT_AS_ID` load option to `true` as shown in https://github.com/epsilonlabs/eugenia-sirius/blob/ce6a75253032c5fc9fc31453c6d2a6d11a93c2f2/plugins/org.eclipse.epsilon.eugenia.sirius/src/org/eclipse/epsilon/eugenia/sirius/models/OdesignModel.java#L11

**Comment by kolovos** [comment, marked as answer]
We have pushed [an example](https://github.com/eclipse/epsilon/tree/main/examples/org.eclipse.epsilon.examples.odesign) that demonstrates loading and processing Sirius Viewpoint Specification Models with EOL.

**Reply by mohamadreza77** [reply]
Thank you, dear Dr. Kolovos. As you kindly mentioned and it is clear in the example, developing within the Eclipse environment and [declaring the ready-to-use Sirius plugins](https://github.com/eclipse/epsilon/tree/main/examples/org.eclipse.epsilon.examples.odesign#:~:text=odesign.setMetamodelUris) added to the project's dependency do the magic.
